pkg/bucket/lifecycle/filter.go

```diff
@@ -38,6 +38,11 @@ type Filter struct {
 // MarshalXML - produces the xml representation of the Filter struct
 // only one of Prefix, And and Tag should be present in the output.
 func (f Filter) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if f.And.isEmpty() && f.Tag.IsEmpty() && f.Prefix == "" {
```

Comment:
As setting empty prefix value is valid, we skip sending empty prefix as per this fix.

XML used in `SetBucketLifecycleConfiguration` API is `<Filter><Prefix></Prefix></Filter>`, with this fix, we don't get back that.

Reply:
AWS doesn't send it back either @balamurugana

Reply:
Ah you are right @balamurugana looks like MinIO server is sending correct values. We don't need this PR.